plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/functions/IcebergFunctionProvider.java

```diff
@@ -40,6 +48,28 @@ public TableFunctionProcessorProviderFactory getTableFunctionProcessorProviderFa
         if (functionHandle instanceof TableChangesFunctionHandle) {
             return new ClassLoaderSafeTableFunctionProcessorProviderFactory(tableChangesFunctionProcessorProviderFactory, getClass().getClassLoader());
         }
+        if (functionHandle instanceof IcebergTablesFunction.IcebergTables) {
+            ClassLoader classLoader = getClass().getClassLoader();
+            return new TableFunctionProcessorProviderFactory()
```

Comment:
Would be nicer to define this as separate class in `io.trino.plugin.iceberg.functions.tables` package similar to `io.trino.plugin.iceberg.functions.tablechanges.TableChangesFunctionProcessorProviderFactory`

Reply:
I'm not sure about this. There is nothing specific to `IcebergTablesFunction` here other than wrapping the creation of `IcebergTablesProcessor` with multiple levels of classloader safe calls.